cylc/flow/cycling/iso8601.py

```diff
@@ -647,111 +666,147 @@ def ingest_time(value, my_now=None):
         # Expanded year
         return value
 
+    value = value.replace(" ", "")  # remove extraneous whitespace
     parser = SuiteSpecifics.point_parser
     offset = None
 
-    if my_now is None:
-        my_now = parser.parse(get_current_time_string())
+    # work out what transformations we need to apply
+    is_prev_next = "next" in value or "previous" in value
+    is_offset = value.startswith("P") or value.startswith("-P")
+    is_truncated = False
+    if not (is_prev_next or is_offset):
+        timepoint = parser.parse(value)
+        is_truncated = timepoint.truncated
+
+    if is_prev_next or is_offset or is_truncated:
+        # we need information about the current time to process this
+        # this should only be attempted if the calendar is in
+        # gregorian mode
+        if CALENDAR.mode != Calendar.MODE_GREGORIAN:
+            raise CylcConfigError(
+                f'The value "{value}" can not be calculated '
+                f'in {CALENDAR.mode}'
+            )
```

Comment:
Is there an equivalent section for https://cylc.github.io/cylc-doc/7.9.1/singlehtml/index.html#scheduling-initial-cycle-point-initial-cycle-point-relative-to-current-time in the cylc8 docs, so that this can be noted there?

Reply:
I'll add a warning to the docs.

Reply:
Documented in a fairly obscure location, these docs are up for review so the important thing for now it that it's in there https://github.com/cylc/cylc-doc/pull/141/files#diff-47b7a9f6c399cc90f13256a3a895c865R1022-R1027